app/src/features/mocks/modals/DeleteCollectionModal/index.tsx

```diff
@@ -0,0 +1,111 @@
+import { DEFAULT_COLLECTION_ID } from "features/mocks/constants";
+import { deleteMocks } from "backend/mocks/deleteMocks";
+import "./deleteCollectionModal.scss";
+
+interface DeleteCollectionModalProps {
+  visible: boolean;
+  collection: RQMockCollection;
+  toggleModalVisibility: (visible: boolean) => void;
+  onSuccess?: () => void;
+}
+
+export const DeleteCollectionModal: React.FC<DeleteCollectionModalProps> = ({
+  visible,
+  collection,
+  toggleModalVisibility,
+  onSuccess,
+}) => {
+  const user = useSelector(getUserAuthDetails);
+  const uid = user?.details?.profile?.uid;
+  const workspace = useSelector(getCurrentlyActiveWorkspace);
+  const teamId = workspace?.id;
+
+  const [isDeletingOnlyCollection, setIsDeletingOnlyCollection] = useState(false);
+  const [isDeletingCollectionAndMocks, setIsDeletingCollectionAndMocks] = useState(false);
+
+  // console.log({ collection });
+
+  const handleDeleteOnlyCollection = async () => {
+    setIsDeletingOnlyCollection(true);
+    const mockIds = collection.children?.map((mock) => mock.id);
+
+    // FIXME: Instead of force rerender, update the local state
+
+    await updateMocksCollectionId(uid, mockIds, DEFAULT_COLLECTION_ID);
+    await updateCollections(uid, [{ id: collection.id, deleted: true }]);
+
+    console.log("Collection deleted!");
+
+    onSuccess?.();
+    message.success("Collection deleted!");
+    setIsDeletingOnlyCollection(false);
+    toggleModalVisibility(false);
+    // force re-render
+  };
+
+  const handleDeleteCollectionAndMocks = async () => {
+    setIsDeletingCollectionAndMocks(true);
+    const mockIds = collection.children?.map((mock) => mock.id);
+
+    await deleteMocks(uid, mockIds, teamId);
+    await updateCollections(uid, [{ id: collection.id, deleted: true }]);
+
+    onSuccess?.();
+    message.success("Collection and mocks deleted!");
+    setIsDeletingOnlyCollection(false);
+    toggleModalVisibility(false);
+  };
+
+  const handleCancel = () => {
+    toggleModalVisibility(false);
+  };
+
+  return (
+    <RQModal
+      width={320}
+      open={visible}
+      destroyOnClose={true}
+      onCancel={handleCancel}
+      className="delete-collection-modal"
+    >
+      <img width={32} height={32} src={deleteIcon} alt="Delete collection" className="icon" />
+      <div className="header">Delete this collection?</div>
+      <div className="description">
+        This will permanently delete this collection. <br />
+        Are you sure you want to delete?
+      </div>
+
+      <div className="actions">
+        <Button
+          block
+          type="default"
+          onClick={handleDeleteOnlyCollection}
+          loading={isDeletingOnlyCollection}
+          disabled={isDeletingCollectionAndMocks}
+        >
+          Keep the mocks
+        </Button>
+        <Button
+          block
+          danger
+          type="primary"
+          onClick={handleDeleteCollectionAndMocks}
+          loading={isDeletingCollectionAndMocks}
```

Comment:
`loading` state is not handled properly. After deleting a collection, the delete modal still shows loading on deleting another collection
![CleanShot 2024-04-30 at 14 50 21](https://github.com/requestly/requestly/assets/16779465/02618e4d-5aa1-4b4d-8e53-c17bd24c19bf)